### ISSUE TITLE
maDLC identity switching- DLC not designed for this?

### ISSUE THREAD

**Issue by fishandfreckles** [DeepLabCut/DeepLabCut#3127]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating System

windows 11

### DeepLabCut version

2.3.9

### What engine are you using?

tensorflow

### DeepLabCut mode

multi animal

### Device type

Geforce 5000 RTX

### Bug description 🐛

Hi- apologies if this isn’t the ideal place to post, but I already shared this question on the image.sc DeepLabCut forum a week ago and haven’t received any replies. I wanted to check whether this might be a limitation of the current multi-animal DeepLabCut, or if I’m missing something.

I’m using DeepLabCut 2.3.9 (multi-animal, TensorFlow) to track ~40–50 fish that are nearly identical in appearance, filmed in the field (ocean footage, somewhat noisy). Each fish is labeled with 3 keypoints (eye, pectoral fin, peduncle) and the training data is hand-labeled to maintain consistent identities across 50 labeled frames of a ~2000-frame video.

Training was done for 200k iterations with dlcrnet_ms5 and identity=false (since the fish are indistinguishable). The training and test errors look good (see below), and loss appears to have plateaued. However, I’m still seeing frequent and large identity switches, even without occlusions. These switches can jump across the frame, and the same fish often disappear or reappear with new IDs. (I've also tried training with pytorch and the results are much much worse, so that is why I've stuck with tensorflow).

I’ve tried tuning all inference parameters and found that raising max_age and pafthreshold helps somewhat, but switches persist. max_age=480 (4 s) currently gives the best results of my plot poses it seems. But still the “create labeled video” output shows constant switching and missing tracks. The Refinement GUI is very difficult to use with 49 individuals with lots of switching, perhaps not feasible. So I’m hoping to understand whether there are other metrics or strategies to stabilize identities, or if DLC wasn't really built for this purpose (maintain identities for large, visually identical groups in the field).

I’d really appreciate any insight into:
-Whether additional labeling would help (given low pixel errors already, I've already tried this once- had 20 frames before and raised to 50- 2.5% of video length)
-Any best practices for minimizing identity switching with visually identical animals (I've already looked online but this is my last ditch effort)
-Whether maDLC is designed to handle this scale of dense, similar individuals in field footage

Thanks so much, I really appreciate any guidance.

### Relevant log output

```shell
(resolution 1920 x 1080)

Results for 200000 training iterations, training fraction of 95, and shuffle 1:
Train error: 4.29 pixels. Test error: 5.05 pixels.
With pcutoff of 0.6:
Train error: 4.29 pixels. Test error: 5.06 pixels.
##########################################
Average Euclidean distance to GT per individual (in pixels; test-only)
individuals
individual1      3.383509
individual10     5.473667
individual11     4.289536
individual12     5.952496
individual13     3.784209
individual14     3.788865
individual15     4.002765
individual16     3.939254
individual17     5.548774
individual18     5.266940
individual19     4.137860
individual2      4.946411
individual20    4.563136
individual21     4.070139
individual22     5.597318
individual23     4.902652
individual24     2.874172
individual25     3.583692
individual26     4.112447
individual27     4.532618
individual28     6.123369
individual29     3.755393
individual3      3.763413
individual30     5.241420
individual31     4.269494
individual32          NaN
individual33          NaN
individual34          NaN
individual35          NaN
individual36          NaN
individual37     5.534900
individual38     3.930715
individual39     3.639788
individual4      3.844542
individual40     5.115355
individual41     5.025668
individual42          NaN
individual43     4.106439
individual5      2.850139
individual6      1.379589
individual7           NaN
individual8      2.555239
individual9      3.917872
Average Euclidean distance to GT per bodypart (in pixels; test-only)
bodyparts
eye         5.822054
pec         4.268264
peduncle    5.011247
```

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by AlexEMG** [comment]
I'd indeed say that maDLC is not optimized for ~50 different fish. 

However, I would recommend trying top-down models and DEKR in the PyTorch 3.x version as that should be much better! There you can also try the CTD version, which is SOTA for crowded scenes: https://openaccess.thecvf.com/content/ICCV2023/papers/Zhou_Rethinking_Pose_Estimation_in_Crowds_Overcoming_the_Detection_Information_Bottleneck_ICCV_2023_paper.pdf

Check out this notebook: https://github.com/DeepLabCut/DeepLabCut/blob/main/examples/COLAB/COLAB_BUCTD_and_CTD_tracking.ipynb
